### PR TITLE
DateTimeV2 entity types should be 2-dimensions since a single entity …

### PR DESCRIPTION
**Pull request by emadelwany**
…might have multiple TIMEX resolutions.

### CODE REVIEW

**Comment by emadelwany** [comment]
@cleemullins @chrimc62 can one of you please review to merge and unblock @gabog?

File: libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs [hunk 1]
```diff
@@ -104,12 +104,12 @@ private static JToken GetEntityValue(EntityRecommendation entity)
 
             if (entity.Type.StartsWith("builtin.datetimeV2."))
             {
-                return new JValue(entity.Resolution?.Values != null && entity.Resolution.Values.Count > 0
-                            ? ((IDictionary<string, object>)((IList<object>)entity.Resolution.Values.First()).First())["timex"]
-                            : entity.Resolution);
+                return entity.Resolution?.Values != null && entity.Resolution.Values.Count > 0
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs]: This code seems very complex. Could you turn it from a single return statement into a nice set of clear code that's more-or-less self documenting?

File: tests/Microsoft.Bot.Builder.LUIS.Tests/LuisRecognizerTests.cs [hunk 2]
```diff
@@ -207,9 +209,35 @@ public class LuisRecognizerTests
             Assert.IsTrue((double)result.Entities["Address"][0]["$instance"]["State"][0]["score"] >= 0 && (double)result.Entities["Address"][0]["$instance"]["State"][0]["score"] <= 1);
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.LUIS.Tests/LuisRecognizerTests.cs]: This line seems very complex and subtle. 

Can you simplify it?

File: libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs (hunk 1 above)
Comment (emadelwany) [reply libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizer.cs]: Sure

File: tests/Microsoft.Bot.Builder.LUIS.Tests/LuisRecognizerTests.cs (hunk 2 above)
Comment (emadelwany) [reply tests/Microsoft.Bot.Builder.LUIS.Tests/LuisRecognizerTests.cs]: Sure...

**Pull request merged**